### PR TITLE
feat: [#280] Implement Create and DropIfExists methods of Schema

### PR DESCRIPTION
**Pull request by hwbrzzl**
## 📑 Description

1. Implement `facades.Schema.Create()` and `facades.Schema.DropIfExists` methods.
2. Implement `table.ID()`, `table.String()`, and `table.Integer()` methods.
3. These methods can be used in a migration file in the next PR.
4. We can support more fields in the future, this is a good start.

Code:

image

Result:

image

## Summary by CodeRabbit

- **New Features**
	- Introduced new interfaces for enhanced database migration management, including `Blueprint`, `ColumnDefinition`, `Grammar`, `Migrator`, and `Repository`.
	- Added methods for creating, dropping, and managing database tables and columns.
	- Enhanced SQL generation and migration operations with new utility functions.
	- Improved schema construction with additional configuration and logging capabilities.

- **Bug Fixes**
	- Improved error handling and logging capabilities within the migration framework.

- **Tests**
	- Added comprehensive test suites for `Blueprint`, `ColumnDefinition`, and other migration functionalities to ensure reliability.

- **Documentation**
	- Enhanced comments and documentation for new methods and interfaces to assist developers in understanding their usage.

## ✅ Checks

- [x] Added test cases for my code

### CODE REVIEW

File: database/migration/blueprint.go
Comment (hwbrzzl) [inline database/migration/blueprint.go:1]: Action: Define various fields, eg: ID,  String, Date, they will be used in a migration file.

File: database/migration/grammars/postgres.go
Comment (hwbrzzl) [inline database/migration/grammars/postgres.go:1]: Action: Different DB drivers have different SQL grammar, they can defined here, each driver has a single file.

File: database/migration/schema.go
Comment (hwbrzzl) [inline database/migration/schema.go:1]: The `facades.Schema` will be called in a migration file:

```
// Up Run the migrations.
func (r *M20240915060148CreateUsersTable) Up() {
	facades.Schema().Create("users", func(table migration.Blueprint) {
		table.ID("id")
		table.String("migration")
		table.Integer("batch")
	})
}

// Down Reverse the migrations.
func (r *M20240915060148CreateUsersTable) Down() {
	facades.Schema().DropIfExists("users")
}
```

**Review by devhaozi — APPROVED** [review]
LGTM

**Comment by tekpriest** [comment]
hey @hwbrzzl really superb feature, going to be released anytime soon?

**Comment by hwbrzzl** [comment]
Yes, will be released in v1.15, still need several months.

**Pull request merged**